docarray/array/array.py

```diff
@@ -182,13 +182,37 @@ def __getitem__(self, item):  # note this should handle slices
         else:
             return self.__list_getitem__(item)
 
+    def __get_stacked_document__(self, item):
+        """
+        In stacked mode this return the Document where the filed stacked are put to None
+        In unstacked mode (default) this is the same as __getitem__
+
+        :param item:
+        :return: Document
+        """
+        if self.is_stacked():
+            doc = super().__getitem__(item)
+            for field in self._columns.keys():
+                setattr(doc, field, None)
+            return doc
+        else:
+            return self.__list_getitem__(item)
+
     def __list_iter__(self):
         return super().__iter__()
 
     def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
+    def __iter_over_stacked_documents__(self):
```

Comment:
yes it should only be called in stacked mode. Technically it could be usefull to call in outside of proto as well maybe in the future.

I will refactor this part anyway lets not do a blocker now technically it will not failed anyway. I really dislike this pattern of fusing both implementation so that is why I want to refactor